### PR TITLE
Failed Tests Log

### PR DESCRIPTION
**Pull request by kuznetsovmoci**
Adding ability to quickly look at the output for failing tests to determine the root cause.
This first view lists all of the platforms containing a failing test.
For each platform, the Brief output of the New Test Matrix is listed for each failing test

### CODE REVIEW

File: scoreboard.pl [hunk 1]
```diff
@@ -539,7 +543,27 @@ ($)
...
+                my $linktarget = '';
+                if (defined $main::opt_n) {
+                    $linktarget = "target=\"_blank\""
+                }
+                # If we have a diff revision, and a diffroot URL, create a link
+                if (($diffRev !~ /None/) && ($diffRoot)) {
+                  my $url = $diffRoot . $diffRev;
+                  $link = "<a href='$url' $linktarget>$diffRev</a>";
+                }
+                Prettify::Process ("$directory/$buildname/$filename", $buildname, $use_build_logs, $link, $directory . "/" . $log_prefix);
```
Comment (iguessthislldo) [inline scoreboard.pl:566]:
```suggestion
                Prettify::Process ("$directory/$buildname/$filename", $buildname, $use_build_logs, $link, "$directory/$log_prefix");
```

File: scoreboard.pl [hunk 2]
```diff
@@ -650,10 +679,30 @@ ($)
...
+                if (defined $main::opt_n) {
+                    $linktarget = "target=\"_blank\""
+                }
+                # If we have a diff revision, and a diffroot URL, create a link
+                if (($diffRev !~ /None/) && ($diffRoot)) {
+                  my $url = $diffRoot . $diffRev;
+                  $link = "<a href='$url' $linktarget>$diffRev</a>";
+                }
+
+                Prettify::Process ("$file.txt", $buildname, $use_build_logs, $link, $directory . "/" . $log_prefix);
```
Comment (iguessthislldo) [inline scoreboard.pl:705]:
```suggestion
                Prettify::Process ("$file.txt", $buildname, $use_build_logs, $link, "$directory/$log_prefix");
```

To be consistent with concatenation vs. interpolation on the same line.

File: common/prettify.pm [hunk 3]
```diff
@@ -353,6 +353,194 @@ sub Normal ($)
...
+sub Print_Sections ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && defined $self->{LAST_SUBSECTION} && $self->{LAST_SECTION} eq 'Test') {
+        if (defined $self->{BUILDNAME}) {
+            print {$self->{FH}} "<hr><h2>$self->{BUILDNAME}</h2>\n";
+            my $rev = substr($self->{GIT_CHECKEDOUT_OPENDDS}, 0, 8);
+            if ($rev ne "unknown") {
+                print {$self->{FH}} "Rev: <a href=$self->{REV_LINK}>$rev</a><hr>\n";
```
Comment (iguessthislldo) [inline common/prettify.pm:480]: Does this link work in your testing?

I added a `<diffroot>` to my test scoreboard file like this:

```xml
    <build>
      <name>dds_doc_Latest_Micro_ant_linux_gcc_d1o0</name>
      <diffroot>https://github.com/objectcomputing/OpenDDS/commit/</diffroot>
    </build>
```

but this link's href ends up blank:

```html
Rev: <a href=>340eb636</a><hr>
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: common/prettify.pm (hunk 3 above)
Comment (iguessthislldo) [reply common/prettify.pm:480]: Also if `REV_LINK` really is empty, we probably shouldn't make it a link.

Comment (kuznetsovmoci) [reply common/prettify.pm:480]: The link is fixed now

Comment (iguessthislldo) [reply common/prettify.pm]: Yup, it's working for me now.

File: common/prettify.pm [hunk 4]
```diff
@@ -353,6 +353,194 @@ sub Normal ($)
...
+sub Print_Sections ()
+{
+    my $self = shift;
+
+    if (defined $self->{LAST_SECTION} && defined $self->{LAST_SUBSECTION} && $self->{LAST_SECTION} eq 'Test') {
+        if (defined $self->{BUILDNAME}) {
+            print {$self->{FH}} "<hr><h2>$self->{BUILDNAME}</h2>\n";
+            my $rev = substr($self->{GIT_CHECKEDOUT_OPENDDS}, 0, 8);
+            if ($rev ne "unknown") {
+                print {$self->{FH}} "Rev: <a href=$self->{REV_LINK}/$rev>$rev</a><hr>\n";
```
Comment (iguessthislldo) [inline common/prettify.pm:480]: Two things: it shouldn't be a link if `diffroot` wasn't set and it the `diffroot` might have already slash at the end like it does on our score board.

```suggestion
                my $rev_line = "Rev: ";
                if (length($self->{REV_LINK})) {
                    $rev_line .= "<a href=$self->{REV_LINK}";
                    $rev_line =~ s/\/$//g;
                    $rev_line .= "/$rev>";
                }
                $rev_line .= $rev;
                if (length($self->{REV_LINK})) {
                    $rev_line .= "</a>";
                }
                print {$self->{FH}} "$rev_line<hr>\n";
```

Other than that everything looks alright.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: common/prettify.pm (hunk 4 above)
Comment (kuznetsovmoci) [reply common/prettify.pm]: Thank you, Fred

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**